### PR TITLE
Update github hook management

### LINKED ISSUE #232: Github force push does not trigger a job
The payload does nos contain `head_commit`, which is what we were using to trigger a job

```
{
  "ref": "refs/heads/web-enter-to-start-job",
  "before": "de6da5757cdc79710d482178dd0861b856f2e95a",
  "after": "0000000000000000000000000000000000000000",
  "created": false,
  "deleted": true,
  "forced": true,
  "base_ref": null,
  "compare": "https://github.com/bazooka-ci/bazooka/compare/de6da5757cdc...000000000000",
  "commits": [

  ],
  "head_commit": null,
  "repository": {
    "id": 26610193,
    "name": "bazooka",
    "full_name": "bazooka-ci/bazooka",
    "owner": {
      "name": "bazooka-ci",
      "email": null
    },
    "private": false,
    "html_url": "https://github.com/bazooka-ci/bazooka",
    "description": "Continuous Integration and Continuous Deployment Server",
    "fork": false,
    "url": "https://github.com/bazooka-ci/bazooka",
    "forks_url": "https://api.github.com/repos/bazooka-ci/bazooka/forks",
    "keys_url": "https://api.github.com/repos/bazooka-ci/bazooka/keys{/key_id}",
    "collaborators_url": "https://api.github.com/repos/bazooka-ci/bazooka/collaborators{/collaborator}",
    "teams_url": "https://api.github.com/repos/bazooka-ci/bazooka/teams",
    "hooks_url": "https://api.github.com/repos/bazooka-ci/bazooka/hooks",
    "issue_events_url": "https://api.github.com/repos/bazooka-ci/bazooka/issues/events{/number}",
    "events_url": "https://api.github.com/repos/bazooka-ci/bazooka/events",
    "assignees_url": "https://api.github.com/repos/bazooka-ci/bazooka/assignees{/user}",
    "branches_url": "https://api.github.com/repos/bazooka-ci/bazooka/branches{/branch}",
    "tags_url": "https://api.github.com/repos/bazooka-ci/bazooka/tags",
    "blobs_url": "https://api.github.com/repos/bazooka-ci/bazooka/git/blobs{/sha}",
    "git_tags_url": "https://api.github.com/repos/bazooka-ci/bazooka/git/tags{/sha}",
    "git_refs_url": "https://api.github.com/repos/bazooka-ci/bazooka/git/refs{/sha}",
    "trees_url": "https://api.github.com/repos/bazooka-ci/bazooka/git/trees{/sha}",
    "statuses_url": "https://api.github.com/repos/bazooka-ci/bazooka/statuses/{sha}",
    "languages_url": "https://api.github.com/repos/bazooka-ci/bazooka/languages",
    "stargazers_url": "https://api.github.com/repos/bazooka-ci/bazooka/stargazers",
    "contributors_url": "https://api.github.com/repos/bazooka-ci/bazooka/contributors",
    "subscribers_url": "https://api.github.com/repos/bazooka-ci/bazooka/subscribers",
    "subscription_url": "https://api.github.com/repos/bazooka-ci/bazooka/subscription",
    "commits_url": "https://api.github.com/repos/bazooka-ci/bazooka/commits{/sha}",
    "git_commits_url": "https://api.github.com/repos/bazooka-ci/bazooka/git/commits{/sha}",
    "comments_url": "https://api.github.com/repos/bazooka-ci/bazooka/comments{/number}",
    "issue_comment_url": "https://api.github.com/repos/bazooka-ci/bazooka/issues/comments{/number}",
    "contents_url": "https://api.github.com/repos/bazooka-ci/bazooka/contents/{+path}",
    "compare_url": "https://api.github.com/repos/bazooka-ci/bazooka/compare/{base}...{head}",
    "merges_url": "https://api.github.com/repos/bazooka-ci/bazooka/merges",
    "archive_url": "https://api.github.com/repos/bazooka-ci/bazooka/{archive_format}{/ref}",
    "downloads_url": "https://api.github.com/repos/bazooka-ci/bazooka/downloads",
    "issues_url": "https://api.github.com/repos/bazooka-ci/bazooka/issues{/number}",
    "pulls_url": "https://api.github.com/repos/bazooka-ci/bazooka/pulls{/number}",
    "milestones_url": "https://api.github.com/repos/bazooka-ci/bazooka/milestones{/number}",
    "notifications_url": "https://api.github.com/repos/bazooka-ci/bazooka/notifications{?since,all,participating}",
    "labels_url": "https://api.github.com/repos/bazooka-ci/bazooka/labels{/name}",
    "releases_url": "https://api.github.com/repos/bazooka-ci/bazooka/releases{/id}",
    "created_at": 1415916203,
    "updated_at": "2015-05-08T15:22:18Z",
    "pushed_at": 1431715436,
    "git_url": "git://github.com/bazooka-ci/bazooka.git",
    "ssh_url": "git@github.com:bazooka-ci/bazooka.git",
    "clone_url": "https://github.com/bazooka-ci/bazooka.git",
    "svn_url": "https://github.com/bazooka-ci/bazooka",
    "homepage": "http://docs.bazooka-ci.io/",
    "size": 4688,
    "stargazers_count": 21,
    "watchers_count": 21,
    "language": "Go",
    "has_issues": true,
    "has_downloads": true,
    "has_wiki": true,
    "has_pages": false,
    "forks_count": 1,
    "mirror_url": null,
    "open_issues_count": 36,
    "forks": 1,
    "open_issues": 36,
    "watchers": 21,
    "default_branch": "master",
    "stargazers": 21,
    "master_branch": "master",
    "organization": "bazooka-ci"
  },
  "pusher": {
    "name": "haklop",
    "email": "eric.bellemon@gmail.com"
  },
  "organization": {
    "login": "bazooka-ci",
    "id": 9402780,
    "url": "https://api.github.com/orgs/bazooka-ci",
    "repos_url": "https://api.github.com/orgs/bazooka-ci/repos",
    "events_url": "https://api.github.com/orgs/bazooka-ci/events",
    "members_url": "https://api.github.com/orgs/bazooka-ci/members{/member}",
    "public_members_url": "https://api.github.com/orgs/bazooka-ci/public_members{/member}",
    "avatar_url": "https://avatars.githubusercontent.com/u/9402780?v=3",
    "description": null
  },
  "sender": {
    "login": "haklop",
    "id": 4128,
    "avatar_url": "https://avatars.githubusercontent.com/u/4128?v=3",
    "gravatar_id": "",
    "url": "https://api.github.com/users/haklop",
    "html_url": "https://github.com/haklop",
    "followers_url": "https://api.github.com/users/haklop/followers",
    "following_url": "https://api.github.com/users/haklop/following{/other_user}",
    "gists_url": "https://api.github.com/users/haklop/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/haklop/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/haklop/subscriptions",
    "organizations_url": "https://api.github.com/users/haklop/orgs",
    "repos_url": "https://api.github.com/users/haklop/repos",
    "events_url": "https://api.github.com/users/haklop/events{/privacy}",
    "received_events_url": "https://api.github.com/users/haklop/received_events",
    "type": "User",
    "site_admin": false
  }
}
```

### PR DESCRIPTION
**Pull request by julienvey**
Now handles

* branches push force
* tags

Closes #232

### CODE REVIEW

File: server/github.go [hunk 1]
```diff
@@ -48,3 +59,30 @@ func (ctx *context) githubAuth(req *http.Request) bool {
...
 }
+
+func (ctx *context) startGithubJob(r *request) (*response, error) {
+	var payload githubPayload
+
+	r.parseBody(&payload)
+
+	var ref string
+	if branchRegexp.MatchString(payload.Ref) {
+		submatch := branchRegexp.FindAllStringSubmatch(payload.Ref, -1)
```
Comment (jawher) [inline server/github.go]: Why do you use the `FindAllStringSubmatch` and not simply `FindStringSubmatch` ?

**Comment by jawher** [comment]
Correct me if I'm mistaken, but with this PR, in the following scenario:

* A user pushes branch `B` at commit `C1`
* Suppose it takes sometime for Github to call the build hook
* The user pushes branch `B` again at commit `C2` (on top of `C1`)

The first hook will start a job with scm ref of `B` and not the precise commit `C1`, and so the triggered build will end-up building `C2`.

**Comment by julienvey** [comment]
Right, I updated the PR

**Comment by jawher** [comment]
LGTM

**Pull request merged**